check/src/completion.rs

```diff
@@ -266,6 +270,17 @@ where
             self.enclosing_match = Match::Pattern(current);
         }
         match current.value {
+            Pattern::As(ref id, ref pat) => {
+                let id_span = Span::new(
+                    current.span.start,
+                    current.span.start + BytePos::from(id.as_ref().len()),
+                );
+                if id_span.containment(&self.pos) == Ordering::Equal {
+                    self.found = Some(Some(Match::Pattern(current)));
```

Comment:
Make this `Match::Ident` instead. That way only the identifier part will be marked in editors.